services/QuillLMS/client/app/bundles/Teacher/startup/CmsUserIndexAppClient.jsx

```diff
@@ -99,6 +100,8 @@ export default class CmsUserIndex extends React.Component {
     }
   };
 
+  toggleAdvanced = () => { this.setState({showAdvanced: !this.state.showAdvanced})};
```

Comment:
If you want to use the current state in a `setState` function, you can pass a function like `this.setState(prevState => ({ showAdvanced: !prevState.showAdvanced }))`. Theoretically that avoids an edge case where the state has changed between the function being invoked and `this.state` gets called. Practically it doesn't really matter in this instance.

Reply:
Ah, nice, I like that pattern, I'll change this.